### PR TITLE
Change default namespace for diagnose

### LINKED ISSUE #119: Meet K8s pod security requirements for subctl diagnose command
**What happened**:
In K8s 1.23 PodSecurity Admission controller is introduced and OCP 4.11 enables this admission controller by default.
While running `subctl diagnose ...` commands command, the following warnings are seen.

```
⠈⡱ Checking Submariner support for the kube-proxy mode W0516 16:30:50.377740   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "query-iface-list" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "query-iface-list" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "query-iface-list" must set securityContext.capabilities.drop=["ALL"]; container "query-iface-list" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "query-iface-list" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "query-iface-list" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking Submariner support for the kube-proxy mode
 ✓ The kube-proxy mode is supported
 
⠈⠑ Checking the firewall configuration to determine if the metrics port (8080) is allowed W0516 16:30:58.959021   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "validate-sniffer" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-sniffer" must set securityContext.capabilities.drop=["ALL"]; container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "validate-sniffer" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-sniffer" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
⢄⡱ Checking the firewall configuration to determine if the metrics port (8080) is allowed W0516 16:31:05.487814   38759 warnings.go:67] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "validate-client" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-client" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "validate-client" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-client" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking the firewall configuration to determine if the metrics port (8080) is allowed
 ✓ The firewall configuration allows metrics to be retrieved from Gateway nodes
 
⠊⠁ Checking the firewall configuration to determine if VXLAN traffic is allowed W0516 16:31:16.307616   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "validate-sniffer" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-sniffer" must set securityContext.capabilities.drop=["ALL"]; container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "validate-sniffer" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-sniffer" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
⢄⡱ Checking the firewall configuration to determine if VXLAN traffic is allowed W0516 16:31:22.879381   38759 warnings.go:67] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "validate-client" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-client" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "validate-client" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-client" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking the firewall configuration to determine if VXLAN traffic is allowed
 ✓ The firewall configuration allows VXLAN traffic
```

**What you expected to happen**:
The warnings should not be displayed. We can modify subctl diagnose to schedule various pods in submariner-operator namespace by default, while retaining the existing option where we allow users to override the namespace if required.

Related to: https://github.com/submariner-io/subctl/issues/27

### PR DESCRIPTION
**Pull request by Jaanki**
diagnose subcommands, kube-proxy-mode and firewall, spawn pods in
"default" ns if it is not provided by the user. For k8s 1.23 and above,
ns without PodSecurity labels throws warnings for the user. It is not
correct to manipulate a ns without the user's consent. Hence change the
"default" ns to "submariner-operator" which is already created with
these labels.

Partially Closes: #119

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Review by skitt — CHANGES_REQUESTED** [review]
You also need to change the namespace flag defaults, _e.g._ https://github.com/submariner-io/subctl/blob/devel/cmd/subctl/diagnose.go#L196

**Comment by Jaanki** [comment]
> You also need to change the namespace flag defaults, _e.g._ https://github.com/submariner-io/subctl/blob/devel/cmd/subctl/diagnose.go#L196

Yes, realised that when I was doing some other related changes. And this got me thinking. Since we are passing a default, the ns would never be nil for diagnose and no need to check for its nil value in internal/pod/schedule.go

**Comment by skitt** [comment]
> > You also need to change the namespace flag defaults, _e.g._ https://github.com/submariner-io/subctl/blob/devel/cmd/subctl/diagnose.go#L196
> 
> Yes, realised that when I was doing some other related changes. And this got me thinking. Since we are passing a default, the ns would never be nil for diagnose and no need to check for its nil value in internal/pod/schedule.go

That’s a belt-and-suspenders approach — it’s better to check for empty values than to have some other change somewhere else cause things to blow up further down the line.

File: pkg/diagnose/firewall.go [hunk 1]
```diff
@@ -130,7 +131,7 @@ func getActiveGatewayNodeName(clusterInfo *cluster.Info, hostname string, imageR
 		// On some platforms, the nodeName does not match with the hostname.
 		// Submariner Endpoint stores the hostname info in the endpoint and not the nodeName. So, we spawn a
 		// tiny pod to read the hostname and return the corresponding node.
-		sPod, err := spawnSnifferPodOnNode(clusterInfo.ClientProducer.ForKubernetes(), node.Name, "default", "hostname",
+		sPod, err := spawnSnifferPodOnNode(clusterInfo.ClientProducer.ForKubernetes(), node.Name, constants.OperatorNamespace, "hostname",
```
Comment (sridhargaddam) [inline pkg/diagnose/firewall.go:134]: We should actually be using the namespace that is read as part of user input (i.e., config.Namespace).

**Review by sridhargaddam — CHANGES_REQUESTED** [review]

File: pkg/diagnose/firewall.go (hunk 1 above)
Comment (Jaanki) [reply pkg/diagnose/firewall.go:134]: The function `getActiveGatewayNodeName` does not take `namespace` as an argument. Its hard-coded. I just changed the hard-coded value.

**Review by sridhargaddam — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**